src/duckling/project/map-select.component.ts

```diff
@@ -0,0 +1,91 @@
+import {Component, ViewContainerRef} from '@angular/core';
+import {MdDialog, MdDialogConfig, MdDialogRef} from '@angular/material';
+import {Observable} from 'rxjs';
+
+import {ProjectService} from './project.service';
+
+/**
+ * Dialog that allows user to choose a map. The result of the dialog is the map key the
+ * user has chosen. The Project service has the logic used to create or load the map.
+ */
+@Component({
+    selector: "dk-map-select",
+    styleUrls: ["./duckling/layout.css"],
+    template: `
+        <div *ngIf="!listLoaded">
+            <md-spinner></md-spinner>
+        </div>
+        <div *ngIf="listLoaded">
+            <dk-section headerText="Select an Existing Map">
+                <md-nav-list>
+                    <md-list-item
+                        *ngFor="let map of maps"
+                        (click)="selectMap(map)">
+                        {{map}}
+                    </md-list-item>
+                </md-nav-list>
+            </dk-section>
+            <dk-section headerText="Create a New Map">
+                <div>
+                    <dk-input class="dk-inline"
+                        label="New Map Name"
+                        [dividerColor]="newMapNameIsValid() ? 'primary' : 'warn'"
+                        (inputChanged)="newMapName = $event">
+                    </dk-input>
+                    <dk-icon-button
+                        icon="save"
+                        tooltip="Create the map"
+                        [disabled]="!newMapNameIsValid()"
+                        (click)="createMap()">
+                    </dk-icon-button>
+                </div>
+            </dk-section>
+        </div>
+    `
+})
+export class MapSelectComponent {
+
+    /**
+     * Open a dialog and return an observable that resolves to the name of the map to open.
+     */
+    static open(viewContainer : ViewContainerRef, currentMap? : string) : Observable<String> {
```

Comment:
you should change this to be a function outside of the class and update the auto-create-animation-component to also use it